pkg/scraper/scraper.go

```diff
@@ -141,17 +143,29 @@ func (c *scraper) Scrape(baseCtx context.Context) *storage.MetricsBatch {
 			responseChannel <- m
 		}(node)
 	}
-
-	res := &storage.MetricsBatch{}
-
+	res := &storage.MetricsBatch{
+		Nodes: make(map[string]storage.NodeMetricsPoint),
+		Pods:  make(map[apitypes.NamespacedName]storage.PodMetricsPoint),
+	}
 	for range nodes {
 		srcBatch := <-responseChannel
 		if srcBatch == nil {
 			continue
 		}
-
-		res.Nodes = append(res.Nodes, srcBatch.Nodes...)
-		res.Pods = append(res.Pods, srcBatch.Pods...)
+		for nodeName, nodeMetricsPoint := range srcBatch.Nodes {
+			if _, nodeFind := res.Nodes[nodeName]; nodeFind {
+				klog.ErrorS(nil, "Got duplicate node point", "node", klog.KRef("", nodeName))
```

Comment:
Since we now check that in the scraper, we can remove the check in the storage as we can expect all the node points to be unique there.